core/evm.go

```diff
@@ -121,14 +123,24 @@ func GetHashFn(ref *types.Header, chain ChainContext) func(n uint64) common.Hash
 	}
 }
 
-// CanTransfer checks whether there are enough funds in the address' account to make a transfer.
+// CanTransfer checks whether there are enough funds in the address' account to make a transfer,
+// and if operation ends up with an overflow.
 // This does not take the necessary gas in to account to make the transfer valid.
-func CanTransfer(db vm.StateDB, addr common.Address, amount *big.Int) bool {
-	return db.GetBalance(addr).Cmp(amount) >= 0
+func CanTransfer(db vm.StateDB, from common.Address, to common.Address, amount *big.Int) error {
+	if db.GetBalance(from).Cmp(amount) < 0 {
+		return vmerrs.ErrInsufficientBalance
+	}
+	// check for overflow. if this is a transfer to self, then we don't need to check for overflow
+	if _, ok := utils.SafeSumUint256(db.GetBalance(to), amount); from != to && !ok {
```

Comment:
can we separate the `from` `to` check so that it is prior to doing the safe sum calculation?

Reply:
we don't need that anymore